sirepo/raydata/analysis_driver/__init__.py

```diff
@@ -78,6 +92,20 @@ def get_run_log(self):
     def has_analysis_pdfs(self):
         return len(self.get_analysis_pdf_paths()) > 0
 
+    def render_papermill_script(self, input_f, output_f):
+        pkjinja.render_resource(
+            _PAPERMILL_SCRIPT,
+            PKDict(
+                input_f=input_f,
+                output_f=output_f,
+                papermill_args=" ".join(self.get_papermill_args()),
+                conda_prefix=self.get_conda_prefix(),
+                conda_env=self.get_conda_env(),
+                catalog_name=self.catalog_name,
+            ),
+            output=_PAPERMILL_SCRIPT,
```

Comment:
This should be `get_papermill_script_path` we shouldn't rely on the fact that the current working dir is the `output_dir` when we call this. Furthermore, this function should return the output path. That way when we call `render_papermill_script` in the scan_monitor it will tell the caller where the script was written to and we can use that instead of calling `d.get_papermill_script_path` again. I think using the return value ties the code together more and reduces chances for bugs.

Reply:
7407ed0e7849051a42a067015819229cd53c406b